store/src/java/com/zimbra/cs/datasource/imap/ImapFolderSync.java

```diff
@@ -100,6 +100,8 @@
     // Max number of total per-item failures before we abort sync
     private static final int MAX_TOTAL_ERRORS = 10;
 
+    private static final String DRAFTS = "Drafts"; 
```

Comment:
I don't see the draft folder loaded up when configured Yahoo as an External Imap account
The same can be seen in attached screenshot
When debugging code this is the list of folders passed as an input to syncFolders/syncMessages

```
inbox=Folder{n=Inbox, id=941, type=1, name=Inbox, unread=1, flags=uy~, folder_id=940, size=1, meta_version=3, version=1, parent_id=940, color=0, subject=Inbox, date=1647409817, rev=8702, deleted=0, del_unread=0, attributes=0}
trash=Folder{n=Trash, id=942, type=1, name=Trash, unread=0, flags=y~, folder_id=940, size=0, meta_version=3, version=1, parent_id=940, color=0, subject=Trash, date=1647409818, rev=8705, deleted=0, del_unread=0, attributes=0}
sent=Folder{n=Sent, id=943, type=1, name=Sent, unread=0, flags=y~, folder_id=940, size=0, meta_version=3, version=1, parent_id=940, color=0, subject=Sent, date=1647409818, rev=8708, deleted=0, del_unread=0, attributes=0}
junk=Folder{n=Junk, id=944, type=1, name=Junk, unread=0, flags=y~, folder_id=940, size=0, meta_version=3, version=1, parent_id=940, color=0, subject=Junk, date=1647409819, rev=8711, deleted=0, del_unread=0, attributes=0}
archive=Folder{n=Archive, id=945, type=1, name=Archive, unread=0, flags=y~, folder_id=940, size=0, meta_version=3, version=1, parent_id=940, color=0, subject=Archive, date=1647409820, rev=8714, deleted=0, del_unread=0, attributes=0}
```

Attching screenshot for same

<img width="1792" alt="Screenshot 2022-03-16 at 11 47 44 AM" src="https://user-images.githubusercontent.com/96303121/158527463-55ef8c33-3991-4466-ba79-57750b3db5b1.png">

When same Yahoo account configured in Outlook Draft Folder is ther and it is named as "Draft"
this code change wouldn't be much if we have Draft folder passed as an input as we can check for "contains" instead of "equals"

Reply:
Relying on `contains` is not ideal either. Yahoo is one of many providers. The special folder can be mapped to different names.

Reply:
Alternatively, see if using the localfolder's id and MailConstants drafts folderId works.

Reply:
The thing is we don't have a draft folder available as an input in the code , and even in UI we don't see draft folder

Reply:
I see that for Yahoo draft folder is ignored 
` <folder remotePath="Draft" ignore="true"/>`
I think, for this reason, the Draft folder for Yahoo is not visible

Reply:
Either way, using the localfolder id will allow it work if the draft folder is being synced (since the mapping would have been done elsewhere) while using the folder path will only work if the path matches?

Reply:
Using Folder path will only allow if the patch matches as of now,
it can be changed to contains if you think it would be appropriate to do it